File: app/src/main/java/com/kakacat/minitool/garbageclassify/Presenter.java
package com.kakacat.minitool.garbageclassify;

import android.text.TextUtils;

import com.kakacat.minitool.common.myinterface.HttpCallback;
import com.kakacat.minitool.common.util.HttpUtil;
import com.kakacat.minitool.common.util.ThreadUtil;
import com.kakacat.minitool.garbageclassify.model.Garbage;
import com.kakacat.minitool.garbageclassify.model.Model;

import java.util.List;

import okhttp3.Response;

public class Presenter implements Contract.Presenter {

    private Contract.View view;
    private Model model;

    public Presenter(Contract.View view) {
        this.view = view;
        this.model = Model.getInstance();
    }

    @Override
    public void initData() {

    }

    @Override
    public void requestData(String s) {
        if (TextUtils.isEmpty(s)) {
            view.onRequestCallBack("请输入内容");
        } else {
            String address = model.getAddress(s);
            HttpUtil.sendOkHttpRequest(address, new HttpCallback() {
                String result = "输入错误或者找不到结果";

                @Override
                public void onSuccess(Response response) {
                    if (model.handleResponse(response)) {
                        result = "处理成功";
                    } else {
                        result = "处理响应字符串失败";
                    }
                    ThreadUtil.callInUiThread(() -> view.onRequestCallBack(result));
                }

                @Override
                public void onError() {
                    ThreadUtil.callInUiThread(() -> view.onRequestCallBack(result));
                }
            });
        }
    }

    @Override
    public List<Garbage> getGarbageList() {
        return model.getGarbageList();
    }
}

File: app/src/main/java/com/kakacat/minitool/garbageclassify/Presenter.kt
package com.kakacat.minitool.garbageclassify

import com.kakacat.minitool.garbageclassify.model.Garbage
import com.kakacat.minitool.garbageclassify.model.Model
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class Presenter(private val view: Contract.View) : Contract.Presenter {

    private val model by lazy { Model() }

    override fun initData() {}

    override fun requestData(s: String?) {
        GlobalScope.launch(Dispatchers.Default) {
            if (!model.validate(s)) {
                view.onRequestCallBack("请输入内容")
            } else {
                val response = model.sendRequest(s!!)
                val handleResult = model.handleResponse(response!!)
                var result = "请求错误"
                if(handleResult){
                    result = if (model.handleResponse(response)) {
                        "处理成功"
                    } else {
                        "处理响应字符串失败"
                    }
                    view.onRequestCallBack(result)
                }else{
                    view.onRequestCallBack(result)
                }
            }
        }
    }

    override val garbageList: List<Garbage>?
        get() = model.garbageList
}